src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -303,21 +313,49 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
                                                                             _lastProfilesNavState,
                                                                             *this);
 
-            contentFrame().Navigate(xaml_typename<Editor::Profiles>(), _lastProfilesNavState);
+            _profileViewModelChangedRevoker = _lastProfilesNavState.Profile().PropertyChanged(winrt::auto_revoke, [=](auto&&, const PropertyChangedEventArgs& args) {
+                const auto settingName{ args.PropertyName() };
+                if (settingName == L"CurrentPage")
+                {
+                    const auto currentPage = _lastProfilesNavState.Profile().CurrentPage();
+                    if (currentPage == L"Base")
+                    {
+                        contentFrame().Navigate(xaml_typename<Editor::Profiles_Base>(), _lastProfilesNavState);
+                        _breadcrumbs.Clear();
+                        _breadcrumbs.Append(RS_(L"Nav_ProfileDefaults/Content"));
+                    }
+                    else if (currentPage == L"Appearance")
+                    {
+                        contentFrame().Navigate(xaml_typename<Editor::Profiles_Appearance>(), _lastProfilesNavState);
+                        _breadcrumbs.Append(RS_(L"Profile_Appearance/Header"));
+                    }
+                    else if (currentPage == L"Advanced")
+                    {
+                        contentFrame().Navigate(xaml_typename<Editor::Profiles_Advanced>(), _lastProfilesNavState);
+                        _breadcrumbs.Append(RS_(L"Profile_Advanced/Header"));
+                    }
+                }
+            });

```

Comment:
This should go into a helper since the exact same thing is done in the other `_Navigate` function, except for the profile name that gets added in the breadcrumb